src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2191,6 +2190,25 @@ def update_azure_service_mesh_profile(self) -> ServiceMeshProfile:
             new_profile.istio.certificate_authority.plugin.cert_chain_object_name = cert_chain_object_name
             updated = True
 
+        # deal with mesh upgrade commands
+        mesh_upgrade_command = self.raw_param.get("mesh_upgrade_command", None)
+        if mesh_upgrade_command is not None:
+            if len(new_profile.istio.revisions) < 2:
+                raise ArgumentUsageError('Azure Service Mesh upgrade is not in progress.')
```

Comment:
would this piece be called during upgrade start as well?

Reply:
No, since mesh_upgrade_command is set to None for upgrade start command